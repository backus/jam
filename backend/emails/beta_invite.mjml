<mjml>
  <mj-body>
    <mj-section>
      <mj-column>
        <mj-text font-size="14px" color="#222222"
          >{{greeting}} I'm reaching out to share your Jam invite with you:
          <a href="{{beta_invite_url}}">{{beta_invite_url}}</a></mj-text
        >

        <mj-text font-size="14px" color="#222222"
          >You joined Jam's waiting list {{time_since_waitlist}} ago, so, in
          case you forgot, Jam is a new service for sharing accounts with
          friends.</mj-text
        >

        <mj-text font-size="14px" color="#222222"
          >Anyways, Jam is very new, so I'm thrilled you took an early interest.
          If you have any feedback, you can reach me at this email. ðŸ˜Š</mj-text
        >

        <mj-text font-size="14px" color="#222222">Thanks,<br />John</mj-text>

        <mj-text font-size="14px" color="#222222"
          >P.S. If you definitely didn't add your email to the waiting list,
          <a style="color: #134394;" href="{{unsubscribe_url}}">click here</a
          >.</mj-text
        >
      </mj-column>
    </mj-section>
  </mj-body>
</mjml>
