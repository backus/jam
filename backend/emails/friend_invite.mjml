<mjml>
  <mj-head>
    <mj-font name="Azo Sans" href="https://use.typekit.net/tet6emj.css" />
  </mj-head>
  <mj-body>
    <mj-section>
      <mj-column>
        <mj-image padding="0" src="{{email_header_url}}" />
      </mj-column>
    </mj-section>
    <mj-section>
      <mj-column>
        <mj-text
          font-size="28px"
          font-weight="900"
          font-family="Azo Sans, Arial, sans-serif"
          align="center"
          color="#555a60"
          >🎁 @{{from_username}} shared a login with you!</mj-text
        >
      </mj-column>
    </mj-section>
    <mj-section>
      <mj-group width="300px">
        <mj-column width="50%">
          <mj-image
            src="{{from_avatar_url}}"
            width="128px"
            padding="0"
            border-radius="100px"
          ></mj-image>
        </mj-column>
        <mj-column width="50%">
          <mj-image
            src="{{share_account_icon_url}}"
            width="128px"
            padding="0"
            border-radius="100px"
          ></mj-image>
        </mj-column>
      </mj-group>
    </mj-section>
    <mj-section>
      <mj-column width="80%">
        <mj-text
          font-size="20px"
          line-height="1.53"
          font-family="Azo Sans, Arial, sans-serif"
          align="center"
          color="#555a60"
          >Jam is a new service for securely sharing passwords with
          friends.</mj-text
        >
      </mj-column>

      <mj-column width="80%">
        <mj-text
          font-size="20px"
          line-height="1.53"
          font-family="Azo Sans, Arial, sans-serif"
          align="center"
          color="#555a60"
          >Your friend <b>@{{from_username}}</b> just gave you access to one of
          their accounts. I wonder what it is...</mj-text
        >
      </mj-column>
    </mj-section>
    <mj-section>
      <mj-column>
        <mj-button
          href="{{invite_url}}"
          height="60px"
          border-radius="5px"
          background-color="#F253CF"
          font-size="24px"
          font-family="Azo Sans, Arial, sans-serif"
          >See what @{{from_username}} shared 🤩</mj-button
        >
      </mj-column>
    </mj-section>
  </mj-body>
</mjml>
