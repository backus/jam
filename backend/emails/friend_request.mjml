<mjml>
  <mj-head>
    <mj-font name="Azo Sans" href="https://use.typekit.net/tet6emj.css" />
  </mj-head>
  <mj-body>
    <mj-section>
      <mj-column>
        <mj-image padding="0" src="{{email_header_url}}" />
      </mj-column>
    </mj-section>
    <mj-section>
      <mj-column>
        <mj-text
          font-size="28px"
          font-weight="900"
          font-family="Azo Sans, Arial, sans-serif"
          align="center"
          color="#555a60"
          >New friend request!</mj-text
        >
      </mj-column>
    </mj-section>
    <mj-section>
      <mj-column>
        <mj-image
          src="{{sender_avatar_url}}"
          width="128px"
          padding="0"
          border-radius="100px"
        ></mj-image>
      </mj-column>
    </mj-section>
    <mj-section>
      <mj-column>
        <mj-text
          font-size="16px"
          font-family="Azo Sans, Arial, sans-serif"
          align="center"
          color="#555a60"
          >@{{sender_username}} wants to be your friend</mj-text
        >
      </mj-column>
    </mj-section>
    <mj-section>
      <mj-column>
        <mj-button
          href="{{notifications_url}}"
          width="300px"
          height="60px"
          border-radius="5px"
          background-color="#F253CF"
          font-size="24px"
          font-family="Azo Sans, Arial, sans-serif"
          >ğŸ’Œ Respond</mj-button
        >
      </mj-column>
    </mj-section>
  </mj-body>
</mjml>
