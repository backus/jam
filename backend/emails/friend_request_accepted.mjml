<mjml>
  <mj-head>
    <mj-font name="Azo Sans" href="https://use.typekit.net/tet6emj.css" />
  </mj-head>
  <mj-body>
    <mj-section>
      <mj-column>
        <mj-image padding="0" src="{{email_header_url}}" />
      </mj-column>
    </mj-section>
    <mj-section>
      <mj-column>
        <mj-text
          font-size="28px"
          font-weight="900"
          font-family="Azo Sans, Arial, sans-serif"
          align="center"
          color="#555a60"
          >ðŸ’• You have a new friend!</mj-text
        >
      </mj-column>
    </mj-section>
    <mj-section>
      <mj-column>
        <mj-image
          src="{{friend_avatar_url}}"
          width="128px"
          padding="0"
          border-radius="100px"
        ></mj-image>
      </mj-column>
    </mj-section>
    <mj-section>
      <mj-column>
        <mj-text
          font-size="16px"
          font-family="Azo Sans, Arial, sans-serif"
          align="center"
          color="#555a60"
          >@{{friend_username}} accepted your friend request. How cute! Go share
          something with @{{friend_username}} now:</mj-text
        >
      </mj-column>
    </mj-section>
    <mj-section>
      <mj-column>
        <mj-button
          href="{{app_url}}"
          height="60px"
          border-radius="5px"
          background-color="#F253CF"
          font-size="24px"
          font-family="Azo Sans, Arial, sans-serif"
          >ðŸ¥° Share with @{{friend_username}}</mj-button
        >
      </mj-column>
    </mj-section>
  </mj-body>
</mjml>
