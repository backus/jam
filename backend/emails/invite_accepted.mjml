<mjml>
  <mj-head>
    <mj-font name="Azo Sans" href="https://use.typekit.net/tet6emj.css" />
  </mj-head>
  <mj-body>
    <mj-section>
      <mj-column>
        <mj-image padding="0" src="{{email_header_url}}" />
      </mj-column>
    </mj-section>
    <mj-section>
      <mj-column>
        <mj-text
          font-size="28px"
          font-weight="900"
          font-family="Azo Sans, Arial, sans-serif"
          align="center"
          color="#555a60"
          >ðŸ™Œ {{invite_nickname}} accepted your invite!</mj-text
        >
      </mj-column>
    </mj-section>
    <mj-section>
      <mj-column>
        <mj-image
          src="{{from_avatar_url}}"
          width="128px"
          padding="0"
          border-radius="100px"
        ></mj-image>
      </mj-column>
    </mj-section>
    <mj-section>
      <mj-column>
        <mj-text
          font-size="16px"
          font-family="Azo Sans, Arial, sans-serif"
          align="center"
          color="#555a60"
          >{{invite_nickname}}'s username is @{{from_username}}. You two are now
          friends on Jam. You should share something with them, I bet it'll make
          them happy:</mj-text
        >
      </mj-column>
    </mj-section>
    <mj-section>
      <mj-column>
        <mj-button
          href="{{app_url}}"
          height="60px"
          border-radius="5px"
          background-color="#F253CF"
          font-size="24px"
          font-family="Azo Sans, Arial, sans-serif"
          >ðŸ˜Š Share with @{{from_username}}</mj-button
        >
      </mj-column>
    </mj-section>
  </mj-body>
</mjml>
