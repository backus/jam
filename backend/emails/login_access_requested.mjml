<mjml>
  <mj-head>
    <mj-font name="Azo Sans" href="https://use.typekit.net/tet6emj.css" />
  </mj-head>
  <mj-body>
    <mj-section>
      <mj-column>
        <mj-image padding="0" src="{{email_header_url}}" />
      </mj-column>
    </mj-section>
    <mj-section>
      <mj-column>
        <mj-text
          font-size="28px"
          font-weight="900"
          font-family="Azo Sans, Arial, sans-serif"
          align="center"
          color="#555a60"
          >ðŸ™‹ @{{from_username}} asked you to share an account on Jam!</mj-text
        >
      </mj-column>
    </mj-section>
    <mj-section>
      <mj-column>
        <mj-image
          src="{{from_avatar_url}}"
          width="128px"
          padding="0"
          border-radius="100px"
        ></mj-image>
      </mj-column>
    </mj-section>
    <mj-section>
      <mj-column width="80%">
        <mj-text
          font-size="16px"
          font-family="Azo Sans, Arial, sans-serif"
          align="center"
          color="#555a60"
          >@{{from_username}} requested access to one of your accounts on Jam.
          You don't have to say yes, but I bet it would make them happy if you
          did.</mj-text
        >
      </mj-column>
    </mj-section>
    <mj-section>
      <mj-column>
        <mj-button
          href="{{notifications_url}}"
          height="60px"
          border-radius="5px"
          background-color="#F253CF"
          font-size="24px"
          font-family="Azo Sans, Arial, sans-serif"
          >Respond to @{{from_username}}</mj-button
        >
      </mj-column>
    </mj-section>
  </mj-body>
</mjml>
