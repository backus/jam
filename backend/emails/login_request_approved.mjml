<mjml>
  <mj-head>
    <mj-font name="Azo Sans" href="https://use.typekit.net/tet6emj.css" />
  </mj-head>
  <mj-body>
    <mj-section>
      <mj-column>
        <mj-image padding="0" src="{{email_header_url}}" />
      </mj-column>
    </mj-section>
    <mj-section>
      <mj-column>
        <mj-text
          font-size="28px"
          font-weight="900"
          font-family="Azo Sans, Arial, sans-serif"
          align="center"
          color="#555a60"
          >ðŸŽ‰ @{{from_username}} shared a login with you!</mj-text
        >
      </mj-column>
    </mj-section>
    <mj-section>
      <mj-group width="300px">
        <mj-column width="50%">
          <mj-image
            src="{{from_avatar_url}}"
            width="128px"
            padding="0"
            border-radius="100px"
          ></mj-image>
        </mj-column>
        <mj-column width="50%">
          <mj-image
            src="{{share_account_icon_url}}"
            width="128px"
            padding="0"
            border-radius="100px"
          ></mj-image>
        </mj-column>
      </mj-group>
    </mj-section>
    <mj-section>
      <mj-column width="80%">
        <mj-text
          font-size="16px"
          font-family="Azo Sans, Arial, sans-serif"
          align="center"
          color="#555a60"
          >@{{from_username}} just gave you access to a login you asked them to
          share. What a good friend!</mj-text
        >
      </mj-column>
    </mj-section>
    <mj-section>
      <mj-column>
        <mj-button
          href="{{notifications_url}}"
          height="60px"
          border-radius="5px"
          background-color="#F253CF"
          font-size="24px"
          font-family="Azo Sans, Arial, sans-serif"
          >View new login</mj-button
        >
      </mj-column>
    </mj-section>
  </mj-body>
</mjml>
