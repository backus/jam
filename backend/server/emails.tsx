import * as React from "react";
import { User, EmailVerification, Invite, WaitlistEntry } from "./models";
import { sendEmail } from "./sendEmail";
import * as path from "path";
import * as fs from "fs";
import globAsync from "glob";
import { promisify } from "util";
import * as _ from "lodash";
import env from "./env";
import { formatDistanceToNow } from "date-fns";
const sanitize = require("xss");
const mjml2html = require("mjml");
const glob = promisify(globAsync);

const emailsDir = path.resolve(env.projectRoot(), "emails");
let templates: { [key: string]: string } = {};

glob(path.resolve(emailsDir, "*.mjml")).then((files) => {
  if (files.length === 0)
    throw new Error(`Couldn't find any email templates in ${emailsDir}!`);

  files.forEach((file) =>
    fs.readFile(file, (_err, data) => {
      const key = path.basename(file).replace(/\.mjml$/, "");
      const { html: compiled } = mjml2html(data.toString());
      templates[key] = compiled;
    })
  );
});

const compileEmail = (name: string) => (locals: { [key: string]: string }) => {
  let email = templates[name];
  if (!email) throw new Error(`No template defined for ${name}!`);

  const matches = [...email.matchAll(/{{(?<name>[\w_]+)}}/g)];
  matches.forEach(([template_name, key]) => {
    const value = locals[key];
    if (!_.isString(value)) throw new Error(`Expected locals to define ${key}`);

    email = email.replace(new RegExp(template_name, "g"), sanitize(value));
  });

  return email;
};

export const verifyEmailHtml = compileEmail("verify_email");

export const sendVerifyEmail = (
  user: User,
  emailVerification: EmailVerification
) => {
  return sendEmail({
    to: user.email,
    subject: "Verify your email",
    text: `Hey ${
      user.username
    }, welcome to Jam!\n\nPlease visit this link to verify your email: ${emailVerification.url()}`,
    html: verifyEmailHtml({
      username: user.username,
      verify_url: emailVerification.url(),
    }),
  });
};

export const friendNotificationEmail = (
  name: string,
  { to, from, data }: { to: User; from: User; data: { [key: string]: any } }
) => {
  switch (name) {
    case "invite_accepted":
      if (!data.inviteNickname)
        throw new Error("Expected nickname from invite to be provided!");
      return sendEmail({
        to: to.email,
        subject: `💕 ${data.inviteNickname} accepted your invite!`,
        text: `Your friend ${
          data.inviteNickname
        } accepted your invite! Their username is @${
          from.username
        }. Share something with them on Jam: ${env.appUrl("/")}`,
        html: compileEmail(name)({
          from_username: from.username,
          from_avatar_url: from.avatarAbsoluteUrl(),
          invite_nickname: data.inviteNickname,
          email_header_url: env.appUrl("email-header.png"),
          app_url: env.appUrl("/"),
        }),
      });
    case "login_access_requested":
      return sendEmail({
        to: to.email,
        subject: `🙋 ${from.username} has a question`,
        text: `@${
          from.username
        } requested access to one of your accounts. You don't have to say yes, but I bet it would make them happy if you did: ${env.appUrl(
          "/notifications"
        )}`,
        html: compileEmail(name)({
          from_username: from.username,
          from_avatar_url: from.avatarAbsoluteUrl(),
          email_header_url: env.appUrl("email-header.png"),
          notifications_url: env.appUrl("/notifications"),
        }),
      });
    case "login_request_approved":
      return sendEmail({
        to: to.email,
        subject: `🎉 ${from.username} shared a login`,
        text: `@${
          from.username
        } just gave you access to a login you asked them to share. Check it out on Jam: ${env.appUrl(
          "/"
        )}`,
        html: compileEmail(name)({
          from_username: from.username,
          from_avatar_url: from.avatarAbsoluteUrl(),
          share_account_icon_url: env.appUrl("account-share-email-icon.png"),
          email_header_url: env.appUrl("email-header.png"),
          notifications_url: env.appUrl("/notifications"),
        }),
      });
    case "new_login_previews":
      if (!("newLoginCount" in data))
        throw new Error("Expected login count to be defined");

      const newLoginCount = data.newLoginCount as number;
      const desc =
        newLoginCount === 1 ? "a new login" : `${newLoginCount} new logins`;
      return sendEmail({
        to: to.email,
        subject: `👀 @${from.username} added ${desc} to Jam!`,
        text: `@${
          from.username
        } added ${desc} new logins to Jam. You can request access here: ${env.appUrl(
          "/"
        )}`,
        html: compileEmail(name)({
          app_url: env.appUrl("/"),
          email_header_url: env.appUrl("email-header.png"),
          from_avatar_url: from.avatarAbsoluteUrl(),
          from_username: from.username,
          logins_desc: desc,
        }),
      });
    case "friend_request_accepted":
      return sendEmail({
        to: to.email,
        subject: `💕 New friend`,
        text: `@${
          from.username
        } accepted your friend request! Share something with them on Jam: ${env.appUrl(
          "/"
        )}`,
        html: compileEmail(name)({
          host: env.domain.toString(),
          friend_username: from.username,
          friend_avatar_url: from.avatarAbsoluteUrl(),
          email_header_url: env.appUrl("email-header.png"),
          app_url: env.appUrl("/"),
        }),
      });
    case "share_login":
      return sendEmail({
        to: to.email,
        subject: `🎁 @${from.username} shared a login!`,
        text: `${
          from.username
        } just shared a new login with you. How exciting! Find out what it is: ${env.appUrl(
          "notifications"
        )}`,
        html: compileEmail(name)({
          host: env.domain.toString(),
          sender_username: from.username,
          sender_avatar_url: from.avatarAbsoluteUrl(),
          email_header_url: env.appUrl("email-header.png"),
          share_account_icon_url: env.appUrl("account-share-email-icon.png"),
          notifications_url: env.appUrl("notifications"),
        }),
      });
    case "friend_request":
      return sendEmail({
        to: to.email,
        subject: `😍 New friend request`,
        text: `${
          from.username
        } wants to be your friend. Visit this link to respond: ${env.appUrl(
          "notifications"
        )}`,
        html: compileEmail(name)({
          host: env.domain.toString(),
          sender_username: from.username,
          sender_avatar_url: from.avatarAbsoluteUrl(),
          email_header_url: env.appUrl("email-header.png"),
          notifications_url: env.appUrl("notifications"),
        }),
      });
    default:
      throw new Error(
        `Not sure what to do with email notification type ${name}`
      );
  }
};

export const friendInviteEmail = (invite: Invite, from: User) => {
  if (!invite.email) throw new Error("This invite doesn't have an emamil!");

  const inviteUrl = env.appUrl(invite.uri());

  return sendEmail({
    to: invite.email,
    subject: `🎁 ${from.username} shared a login with you!`,
    text:
      `@${from.username} just shared a login with you using Jam! ` +
      `Jam is a new service for securely sharing passwords with friends. ` +
      `Go here to find out what they gave you: ${inviteUrl}`,
    html: compileEmail("friend_invite")({
      email_header_url: env.appUrl("email-header.png"),
      from_username: from.username,
      from_avatar_url: from.avatarAbsoluteUrl(),
      share_account_icon_url: env.appUrl("account-share-email-icon.png"),
      invite_url: inviteUrl,
    }),
  });
};

export const betaInviteEmail = (waitlistEntry: WaitlistEntry) => {
  const inviteUrl = env.appUrl(waitlistEntry.betaInviteUri());
  const unsubscribeUrl = env.appUrl(waitlistEntry.unsubscribeUri());

  const greeting = waitlistEntry.firstName
    ? `Hi ${waitlistEntry.firstName}!`
    : "Hi!";

  const timeSinceWaitlist = formatDistanceToNow(waitlistEntry.createdAt);

  return sendEmail({
    from: "Jam <support@jam.link>",
    to: waitlistEntry.email,
    subject: `Jam beta invite`,
    text: [
      `${greeting} I'm reaching out to share your Jam invite with you: ${inviteUrl}`,
      "",
      `You joined Jam's waiting list ${timeSinceWaitlist} ago, so, in case you forgot, Jam is a new service for sharing accounts with friends.`,
      "",
      "Anyways, Jam is very new, so I'm thrilled you took an early interest. If you have any feedback, you can reach me at this email.",
      "",
      "Thanks,",
      "John",
      "",
      `P.S. If you definitely didn't add your email to the waiting list, you can unsubscribe here: ${unsubscribeUrl}`,
    ].join("\n"),
    html: compileEmail("beta_invite")({
      beta_invite_url: inviteUrl,
      greeting,
      time_since_waitlist: timeSinceWaitlist,
      unsubscribe_url: unsubscribeUrl,
    }),
  });
};

export const waitlistVerificationEmail = (entry: WaitlistEntry) => {
  const verificationUrl = env.appUrl(entry.verifyEmailUri());

  return sendEmail({
    to: entry.email,
    subject: "Confirm your email",
    text: `Hi ${
      entry.firstName || "there"
    }. Click this link to confirm your email and join the Jam waiting list: ${verificationUrl}`,
    html: compileEmail("verify_waitlist_email")({
      email_header_url: env.appUrl("email-header.png"),
      first_name: entry.firstName || "there",
      verification_url: verificationUrl,
    }),
  });
};
