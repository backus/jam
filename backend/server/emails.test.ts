import { verifyEmailHtml } from "./emails";

it("generates an email template", () => {
  expect(
    verifyEmailHtml({
      username: "specialuser",
      verify_url: "https://jam.link/verify/123",
    })
  ).toMatchSnapshot();
});

it("throws an error if we don't complete all the templates", () => {
  expect(() =>
    verifyEmailHtml({
      username: "specialuser",
    })
  ).toThrowError("Expected locals to define verify_url");
});
