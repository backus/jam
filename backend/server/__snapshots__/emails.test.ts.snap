// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generates an email template 1`] = `
"
    <!doctype html>
    <html xmlns=\\"http://www.w3.org/1999/xhtml\\" xmlns:v=\\"urn:schemas-microsoft-com:vml\\" xmlns:o=\\"urn:schemas-microsoft-com:office:office\\">
      <head>
        <title>
          
        </title>
        <!--[if !mso]><!-- -->
        <meta http-equiv=\\"X-UA-Compatible\\" content=\\"IE=edge\\">
        <!--<![endif]-->
        <meta http-equiv=\\"Content-Type\\" content=\\"text/html; charset=UTF-8\\">
        <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1\\">
        <style type=\\"text/css\\">
          #outlook a { padding:0; }
          body { margin:0;padding:0;-webkit-text-size-adjust:100%;-ms-text-size-adjust:100%; }
          table, td { border-collapse:collapse;mso-table-lspace:0pt;mso-table-rspace:0pt; }
          img { border:0;height:auto;line-height:100%; outline:none;text-decoration:none;-ms-interpolation-mode:bicubic; }
          p { display:block;margin:13px 0; }
        </style>
        <!--[if mso]>
        <xml>
        <o:OfficeDocumentSettings>
          <o:AllowPNG/>
          <o:PixelsPerInch>96</o:PixelsPerInch>
        </o:OfficeDocumentSettings>
        </xml>
        <![endif]-->
        <!--[if lte mso 11]>
        <style type=\\"text/css\\">
          .mj-outlook-group-fix { width:100% !important; }
        </style>
        <![endif]-->
        
      <!--[if !mso]><!-->
        <link href=\\"https://fonts.googleapis.com/css?family=Ubuntu:300,400,500,700\\" rel=\\"stylesheet\\" type=\\"text/css\\">
        <style type=\\"text/css\\">
          @import url(https://fonts.googleapis.com/css?family=Ubuntu:300,400,500,700);
        </style>
      <!--<![endif]-->

    
        
    <style type=\\"text/css\\">
      @media only screen and (min-width:480px) {
        .mj-column-per-100 { width:100% !important; max-width: 100%; }
      }
    </style>
    
  
        <style type=\\"text/css\\">
        
        
        </style>
        
        
      </head>
      <body>
        
        
      <div
         style=\\"\\"
      >
        
      
      <!--[if mso | IE]>
      <table
         align=\\"center\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" class=\\"\\" style=\\"width:600px;\\" width=\\"600\\"
      >
        <tr>
          <td style=\\"line-height:0px;font-size:0px;mso-line-height-rule:exactly;\\">
      <![endif]-->
    
      
      <div  style=\\"background:#877EEF;background-color:#877EEF;margin:0px auto;max-width:600px;\\">
        
        <table
           align=\\"center\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" role=\\"presentation\\" style=\\"background:#877EEF;background-color:#877EEF;width:100%;\\"
        >
          <tbody>
            <tr>
              <td
                 style=\\"direction:ltr;font-size:0px;padding:20px 0;text-align:center;\\"
              >
                <!--[if mso | IE]>
                  <table role=\\"presentation\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\">
                
        <tr>
      
            <td
               class=\\"\\" style=\\"vertical-align:top;width:600px;\\"
            >
          <![endif]-->
            
      <div
         class=\\"mj-column-per-100 mj-outlook-group-fix\\" style=\\"font-size:0px;text-align:left;direction:ltr;display:inline-block;vertical-align:top;width:100%;\\"
      >
        
      <table
         border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" role=\\"presentation\\" style=\\"vertical-align:top;\\" width=\\"100%\\"
      >
        
            <tr>
              <td
                 align=\\"center\\" style=\\"font-size:0px;padding:10px 25px;word-break:break-word;\\"
              >
                
      <div
         style=\\"font-family:Arial, sans-serif;font-size:32px;font-weight:bold;line-height:50px;text-align:center;color:#ffffff;\\"
      >ðŸ‘‹ Ready to Jam?</div>
    
              </td>
            </tr>
          
      </table>
    
      </div>
    
          <!--[if mso | IE]>
            </td>
          
        </tr>
      
                  </table>
                <![endif]-->
              </td>
            </tr>
          </tbody>
        </table>
        
      </div>
    
      
      <!--[if mso | IE]>
          </td>
        </tr>
      </table>
      
      <table
         align=\\"center\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" class=\\"\\" style=\\"width:600px;\\" width=\\"600\\"
      >
        <tr>
          <td style=\\"line-height:0px;font-size:0px;mso-line-height-rule:exactly;\\">
      <![endif]-->
    
      
      <div  style=\\"margin:0px auto;max-width:600px;\\">
        
        <table
           align=\\"center\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" role=\\"presentation\\" style=\\"width:100%;\\"
        >
          <tbody>
            <tr>
              <td
                 style=\\"direction:ltr;font-size:0px;padding:20px 0;text-align:center;\\"
              >
                <!--[if mso | IE]>
                  <table role=\\"presentation\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\">
                
        <tr>
      
            <td
               class=\\"\\" style=\\"vertical-align:top;width:600px;\\"
            >
          <![endif]-->
            
      <div
         class=\\"mj-column-per-100 mj-outlook-group-fix\\" style=\\"font-size:0px;text-align:left;direction:ltr;display:inline-block;vertical-align:top;width:100%;\\"
      >
        
      <table
         border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" role=\\"presentation\\" style=\\"vertical-align:top;\\" width=\\"100%\\"
      >
        
            <tr>
              <td
                 align=\\"left\\" style=\\"font-size:0px;padding:10px 25px;padding-top:30px;word-break:break-word;\\"
              >
                
      <div
         style=\\"font-family:Ubuntu, Helvetica, Arial, sans-serif;font-size:16px;line-height:1;text-align:left;color:#000000;\\"
      >Hey @specialuser, ready to Jam? Click here to verify your email:</div>
    
              </td>
            </tr>
          
      </table>
    
      </div>
    
          <!--[if mso | IE]>
            </td>
          
        </tr>
      
                  </table>
                <![endif]-->
              </td>
            </tr>
          </tbody>
        </table>
        
      </div>
    
      
      <!--[if mso | IE]>
          </td>
        </tr>
      </table>
      
      <table
         align=\\"center\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" class=\\"\\" style=\\"width:600px;\\" width=\\"600\\"
      >
        <tr>
          <td style=\\"line-height:0px;font-size:0px;mso-line-height-rule:exactly;\\">
      <![endif]-->
    
      
      <div  style=\\"margin:0px auto;max-width:600px;\\">
        
        <table
           align=\\"center\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" role=\\"presentation\\" style=\\"width:100%;\\"
        >
          <tbody>
            <tr>
              <td
                 style=\\"direction:ltr;font-size:0px;padding:20px 0;text-align:center;\\"
              >
                <!--[if mso | IE]>
                  <table role=\\"presentation\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\">
                
        <tr>
      
            <td
               class=\\"\\" style=\\"vertical-align:top;width:600px;\\"
            >
          <![endif]-->
            
      <div
         class=\\"mj-column-per-100 mj-outlook-group-fix\\" style=\\"font-size:0px;text-align:left;direction:ltr;display:inline-block;vertical-align:top;width:100%;\\"
      >
        
      <table
         border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" role=\\"presentation\\" style=\\"vertical-align:top;\\" width=\\"100%\\"
      >
        
            <tr>
              <td
                 align=\\"center\\" vertical-align=\\"middle\\" style=\\"font-size:0px;padding:10px 25px;padding-top:10px;padding-bottom:45px;word-break:break-word;\\"
              >
                
      <table
         border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" role=\\"presentation\\" style=\\"border-collapse:separate;line-height:100%;\\"
      >
        <tr>
          <td
             align=\\"center\\" bgcolor=\\"#F253CF\\" role=\\"presentation\\" style=\\"border:none;border-radius:5px;cursor:auto;mso-padding-alt:10px 25px;background:#F253CF;\\" valign=\\"middle\\"
          >
            <a
               href=\\"https://jam.link/verify/123\\" style=\\"display:inline-block;background:#F253CF;color:#ffffff;font-family:Arial, sans-serif, Helvetica, Arial, sans-serif;font-size:24px;font-weight:bold;line-height:120%;margin:0;text-decoration:none;text-transform:none;padding:10px 25px;mso-padding-alt:0px;border-radius:5px;\\" target=\\"_blank\\"
            >
              Verify me!!
            </a>
          </td>
        </tr>
      </table>
    
              </td>
            </tr>
          
      </table>
    
      </div>
    
          <!--[if mso | IE]>
            </td>
          
        </tr>
      
                  </table>
                <![endif]-->
              </td>
            </tr>
          </tbody>
        </table>
        
      </div>
    
      
      <!--[if mso | IE]>
          </td>
        </tr>
      </table>
      <![endif]-->
    
    
      </div>
    
      </body>
    </html>
  "
`;
